amr-wind/core/FieldFillPatchOps.H

```diff
@@ -270,13 +273,34 @@ public:
     {
 
         if (lev == 0) {
-            amrex::PhysBCFunct<amrex::GpuBndryFuncFab<Functor>> physbc(
-                m_mesh.Geom(lev), bcrec, bc_functor());
+            amrex::PhysBCFunct<amrex::GpuBndryFuncFab<FunctorX>> physbcx(
+                m_mesh.Geom(lev), bcrec, bc_functor_x());
```

Comment:
What would be neat here is to keep the for loop and instead do `bc_functor_f(i)`. Or maybe it becomes easier with the templating.

Reply:
I'm having a hard time getting the loop to work. The problem is that the type of physbcx depends on the direction, in addition to the bc_functor type. I have templatized the FunctorF type, but the template argument has to be a const expression, so it cannot be according to the index i. Making bc_functor output different things depending on an input argument is a problem, too, because each output would be a different type. I also can't put the physbcs into a vector because they're different types. Not sure if there is a way forward here.